src/item-detail/item-detail.js

```diff
@@ -35,18 +35,37 @@ class ItemDetail extends Component {
     constructor(props) {
         super(props);
 
-        this.state = {...initState};
+        this.state = {
+            ...initState,
+            description: props.params.description
+        };
 
         this.onChangeTypeOfContent = this.onChangeTypeOfContent.bind(this);
+        this.onUpdateDescription = this.onUpdateDescription.bind(this);
+        this.onSaveDescription = this.onSaveDescription.bind(this);
     }
 
     onChangeTypeOfContent(isEditable) {
         this.setState({isEditable})
     }
 
+    onUpdateDescription(newDescription) {
+        this.setState({
+            description: newDescription
+        })
+    }
+
+    onSaveDescription(){
+        this.props.onChangeDescription(this.state.description);
```

Comment:
that is it? you are pass the args - aka e.target.value is description, so why do you insert new property to the state? is it necessary? that the problem to use it directly?

Reply:
Since we have the controlled input, `state.description` will always have the updated description. We can pass `e.target.value` instead, if it makes it more readable! That should not be a problem.